sycl/source/detail/device_impl.hpp

```diff
@@ -589,33 +589,27 @@ class device_impl : public std::enable_shared_from_this<device_impl> {
     }
 
     CASE(info::device::usm_device_allocations) {
-      return get_info_impl_nocheck<UR_DEVICE_INFO_USM_DEVICE_SUPPORT>()
-                 .value_or(0) &
+      return get_info_impl<UR_DEVICE_INFO_USM_DEVICE_SUPPORT>() &
```

Comment:
@intel/unified-runtime-reviewers , are any of these APIs expected to return a meaningful error that isn't an indication of everything being terribly broken?

Reply:
no, these aren't optional queries so it should be success or catastrophic failure as you say